src/Distributions.cpp

```diff
@@ -25,18 +25,25 @@ DISTRIBUTION(NormMean,
 )
 
 
-DISTRIBUTION(NormVar,
+DISTRIBUTION(var_norm,
     double costFunction(int start, int end){
-        return 5;
+        double lSum = this -> cumsum -> getLinearSum(start, end);
+        double sSum =  this -> cumsum -> getQuadraticSum(start, end);
+        int N = end - start + 1;
+        double mean = this -> cumsum -> getTotalMean(); // Fixed mean
+        double varN = (sSum - 2 * mean * lSum + N * pow(mean, 2)); // Variance of segment.
+        if(varN <= 0){  // Making sure the variance is not zero, so that the log does not raise an exception
+            varN = 0.00000000001;
```

Comment:
@diego-urgell please avoid using abritrary numeric constants like this. what was your intent here?
you can maybe just return INFINITY from math.h instead of calling log?

Reply:
Yes! That's a great idea and it makes it easier to check what you mentioned before about returning zero possible change points if every segment has zero variance (and therefore infinite cost)